src/ir/proto/sql/decoder_context_test.cc

```diff
@@ -39,4 +39,30 @@ TEST(IRContextTest, GetOrCreateStorageIdempotence) {
             types::TypeBase::Kind::kPrimitive);
 }
 
+TEST(DecoderContextTest, DecoderContextRegisterAndGetValueTest) {
+  IRContext ir_context;
+  DecoderContext decoder_context(ir_context);
+
+  Storage storage("storage1", ir_context.type_factory().MakePrimitiveType());
+
+  const Value &value1 = decoder_context.RegisterValue(1, Value(value::Any()));
+  EXPECT_THAT(value1.If<value::Any>(), testing::NotNull());
+  const Value &value2 =
+      decoder_context.RegisterValue(2, Value(value::StoredValue(storage)));
+  const Value &get_value1 = decoder_context.GetValue(1);
+  const Value &get_value2 = decoder_context.GetValue(2);
+  EXPECT_EQ(&get_value1, &value1);
```

Comment:
Not related to this PR, but we should add a comparator for `Value`.

Reply:
Agreed.